### PR TITLE
feat: Integrate Consumer group actions items

### PR DESCRIPTION
**Pull request by hemahg**
**What this PR does / why we need it**:

- [x]  Integrate consumer group delete modal
- [x]  Integrate consumer group view partition Drawer
- [x] Integrate consumer group reset offset modal

**Which issue(s) this PR fixes** 

> Replace this comment with references to related issues.
> e.g. `fixes #<issue number>`

**Verification steps** 

> Attach a screenshot of the output.
> guidance on how to go the particular story.
> e.g. `point to the story with a text breadcrumb, something like "components > <component name> > <story name>”`

**Is there any work left / what's next** :

> Replace this comment with pending items that need to be covered in this PR or some pending items you are planning to do in the future PR.

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by hemahg** [comment]
Hi @riccardo-forina, I created a gate to check the consumer group inside the topic table
https://github.com/redhat-developer/rhosak-ui/blob/80437865167666d97464c3ccbf7e4e6dbfeed227/apps/consoledot-rhosak/src/routes/data-plane/UseTopicConsumerGroupGate.tsx and using this to get details for consumer group action item inside the topic, could you please review this. I wanted to make sure if it's a better approach

File: apps/consoledot-rhosak/src/routes/data-plane/UseTopicConsumerGroupGate.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    true
+  );
+
+  return {
+    instance,
+    consumerGroup: consumerGroup as NonNullable<typeof consumerGroup>,
+    match,
+    topic,
+  };
+}
```
Comment (riccardo-forina) [inline apps/consoledot-rhosak/src/routes/data-plane/UseTopicConsumerGroupGate.tsx:32]: `useTopicConsumerGroupGate.tsx`, lowercase `u`.

I'm on the fence about these two hooks, as they really are very similar. But as I said I think some duplication here is better than an abstraction.

File: apps/consoledot-rhosak/src/routes/data-plane/routes/TopicConsumerGroupsRoute.tsx [hunk 2]
```diff
@@ -51,6 +54,36 @@ export const TopicConsumerGroupsRoute: VoidFunctionComponent<
     topic: topic.name,
   });
 
+  const onDeletConsumerGroup = useCallback(
+    (groupId: string) => {
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-call,@typescript-eslint/no-unsafe-member-access
+      history.push(
+        `${ControlPlaneRouteRoot}/${instance.id}/details/topics/${topic.name}/consumer-groups/${groupId}/delete`
```
Comment (riccardo-forina) [inline apps/consoledot-rhosak/src/routes/data-plane/routes/TopicConsumerGroupsRoute.tsx:61]: As above. This shouldn't be hardcoded. Maybe add a TODO to fix this later.

File: apps/consoledot-rhosak/src/routes/data-plane/routes/TopicConsumerGroupsRoute.tsx [hunk 3]
```diff
@@ -51,6 +54,36 @@ export const TopicConsumerGroupsRoute: VoidFunctionComponent<
...
+      );
+    },
+    [history, instance.id, topic.name]
+  );
+
+  const onViewPartition = useCallback(
+    (groupId: string) => {
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-call,@typescript-eslint/no-unsafe-member-access
+      history.push(
+        `${ControlPlaneRouteRoot}/${instance.id}/details/topics/${topic.name}/consumer-groups/${groupId}/view-partition`
```
Comment (riccardo-forina) [inline apps/consoledot-rhosak/src/routes/data-plane/routes/TopicConsumerGroupsRoute.tsx:71]: As above. This shouldn't be hardcoded. Maybe add a TODO to fix this later.

File: apps/consoledot-rhosak/src/routes/data-plane/routes/TopicConsumerGroupsRoute.tsx [hunk 4]
```diff
@@ -51,6 +54,36 @@ export const TopicConsumerGroupsRoute: VoidFunctionComponent<
...
+      );
+    },
+    [history, instance.id, topic.name]
+  );
+
+  const onClickResetOffset = useCallback(
+    (groupId: string) => {
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-call,@typescript-eslint/no-unsafe-member-access
+      history.push(
+        `${ControlPlaneRouteRoot}/${instance.id}/details/topics/${topic.name}/consumer-groups/${groupId}/reset-offset`
```
Comment (riccardo-forina) [inline apps/consoledot-rhosak/src/routes/data-plane/routes/TopicConsumerGroupsRoute.tsx:81]: As above. This shouldn't be hardcoded. Maybe add a TODO to fix this later.

File: packages/consoledot-api/src/queries/useConsumerGroup.ts [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+  return useQuery({
+    queryKey: kafkaQueries.instance.consumerGroup(params),
+    queryFn: () => {
+      if (!params.id) {
+        return Promise.reject("Invalid id");
+      }
+      if (!params.adminUrl) {
+        return Promise.reject("Invalid adminUrl");
+      }
```
Comment (riccardo-forina) [inline packages/consoledot-api/src/queries/useConsumerGroup.ts:26]:
```suggestion
      if (!params.adminUrl) {
        return Promise.reject("Invalid adminUrl");
      }
      if (!params.consumerGroupId) {
        return Promise.reject("Invalid consumerGroupId");
      }
      
```

File: packages/ui/src/components/ConsumerGroups/components/ConsumerGroupResetOffset.tsx [hunk 6]
```diff
@@ -98,13 +93,47 @@ export const ConsumerGroupResetOffset: FunctionComponent<
     new_offset: t("consumerGroup.new_offset"),
   };
 
+  const onSelectAllConsumer = (isSelecting = true) => {
+    setSelectableConsumer(
+      selectableConsumer.map((consumer) => {
+        consumer.selected = isSelecting;
```
Comment (riccardo-forina) [inline packages/ui/src/components/ConsumerGroups/components/ConsumerGroupResetOffset.tsx:99]: Never mutate a data structure, especially one that represents data coming from an external source! Please keep track of the selected items in a different state.

https://javascript.plainenglish.io/javascript-mutable-vs-immutable-1efb662d78c8

**Review by riccardo-forina — CHANGES_REQUESTED** [review]
Overall it looks good, but I had to flag a mistake where you mutate a data structure in-place. Always consider our data as an immutable data structure

File: packages/ui/src/components/ConsumerGroups/components/ConsumerGroupResetOffset.tsx [hunk 7]
```diff
@@ -18,74 +18,71 @@ import {
...
 } from "@patternfly/react-table";
-import type { FunctionComponent } from "react";
+import { useCallback, useEffect, useState } from "react";
 import { useTranslation } from "react-i18next";
 import type { Consumer } from "ui-models/src/models/consumer-group";
 import "../ConsumerGroup.css";
 import type { OffsetValue } from "../types";
 import { OffsetSelect } from "./OffsetSelect";
 import { TopicSelect } from "./TopicSelect";
+import type { ConsumerGroupState } from "@rhoas/app-services-ui-components";
```
Comment (riccardo-forina) [inline packages/ui/src/components/ConsumerGroups/components/ConsumerGroupResetOffset.tsx:28]: Don't import from ui-components, all the Kafka related stuff from there will be removed eventually. You should have this as `ConsumerGroup['state']` with `ConsumerGroup` imported from `ui-models` (I think).

File: apps/consoledot-rhosak/src/routes/data-plane/routes/TopicConsumerGroupsRoute.tsx (hunk 2 above)
Comment (hemahg) [reply apps/consoledot-rhosak/src/routes/data-plane/routes/TopicConsumerGroupsRoute.tsx:64]: I have added a TODO for now

File: apps/consoledot-rhosak/src/routes/data-plane/routes/ConsumerGroupsRoute.tsx [hunk 8]
```diff
@@ -46,6 +51,36 @@ export const ConsumerGroupsRoute: VoidFunctionComponent<
     direction: sortDirection,
     groupId: consumerName.chips[0],
   });
+
+  const onDeletConsumerGroup = useCallback(
+    (groupId: string) => {
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-call,@typescript-eslint/no-unsafe-member-access
+      history.push(
+        `${instanceConsumerGroupsHref(instance.id)}/${groupId}/delete`
```
[the remark this answers is not in the document]
Comment (hemahg) [reply apps/consoledot-rhosak/src/routes/data-plane/routes/ConsumerGroupsRoute.tsx:61]: I have added a TODO for now

File: apps/consoledot-rhosak/src/routes/data-plane/routes/ConsumerGroupsRoute.tsx [hunk 9]
```diff
@@ -46,6 +51,36 @@ export const ConsumerGroupsRoute: VoidFunctionComponent<
...
+      );
+    },
+    [history, instance.id]
+  );
+
+  const onClickResetOffset = useCallback(
+    (groupId: string) => {
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-call,@typescript-eslint/no-unsafe-member-access
+      history.push(
+        `${instanceConsumerGroupsHref(instance.id)}/${groupId}/reset-offset`
```
[the remark this answers is not in the document]
Comment (hemahg) [reply apps/consoledot-rhosak/src/routes/data-plane/routes/ConsumerGroupsRoute.tsx:83]: I have added a TODO for now

File: apps/consoledot-rhosak/src/routes/data-plane/routes/TopicConsumerGroupsRoute.tsx (hunk 3 above)
Comment (hemahg) [reply apps/consoledot-rhosak/src/routes/data-plane/routes/TopicConsumerGroupsRoute.tsx:75]: I have added a TODO for now

File: apps/consoledot-rhosak/src/routes/data-plane/routes/TopicConsumerGroupsRoute.tsx (hunk 4 above)
Comment (hemahg) [reply apps/consoledot-rhosak/src/routes/data-plane/routes/TopicConsumerGroupsRoute.tsx:86]: I have added a TODO for now

File: packages/consoledot-api/src/queries/useConsumerGroup.ts (hunk 5 above)
Comment (hemahg) [reply packages/consoledot-api/src/queries/useConsumerGroup.ts:26]: updated as per the comment

File: packages/ui/src/components/ConsumerGroups/components/ConsumerGroupResetOffset.tsx (hunk 6 above)
Comment (hemahg) [reply packages/ui/src/components/ConsumerGroups/components/ConsumerGroupResetOffset.tsx]: Done

File: packages/ui/src/components/ConsumerGroups/components/ConsumerGroupResetOffset.tsx [hunk 10]
```diff
@@ -18,74 +18,71 @@ import {
...
+  consumers,
+  state,
 }) => {
   const { t } = useTranslation(["kafka"]);
 
+  const [selectedTopic, setSelectedTopic] = useState<string>("");
+  const [selectedOffset, setSelectedOffset] = useState<OffsetValue>("absolute");
+  const [customOffsetValue, setcustomOffsetValue] = useState<string>();
+
+  const [isCheckboxChecked, setIsCheckboxChecked] = useState<boolean>(false);
```
Comment (riccardo-forina) [inline packages/ui/src/components/ConsumerGroups/components/ConsumerGroupResetOffset.tsx:71]:
```suggestion
  const [isResetAccepted, setIsResetAccepted] = useState<boolean>(false);
```

I think this makes it a bit more clear what this state is for, what do you think?

**Review by riccardo-forina — CHANGES_REQUESTED** [review]
I quickly went through the last changes and spotted a few things.

File: packages/ui/src/components/ConsumerGroups/components/ConsumerGroupResetOffset.tsx (hunk 10 above)
Comment (hemahg) [reply packages/ui/src/components/ConsumerGroups/components/ConsumerGroupResetOffset.tsx:71]: yeah, I agree

Comment (hemahg) [reply packages/ui/src/components/ConsumerGroups/components/ConsumerGroupResetOffset.tsx]: changes as the review comment

**Review by riccardo-forina — APPROVED** [review]

**Pull request merged**